pkg/driver/nydus/nydus.go

```diff
@@ -433,76 +436,124 @@ func (d *Driver) getChunkDict(ctx context.Context, provider accelcontent.Provide
+	if err != nil {
+		return nil, errors.Wrap(err, "marshal remote cache to manifest index  failed")
+	}
+	if err = content.WriteBlob(ctx, pvd.ContentStore(), ref, bytes.NewReader(manifestIndexBytes), *manifestIndexDesc); err != nil {
+		return nil, errors.Wrap(err, "remote cache write blob failed")
 	}
 
-	// TODO: handle manifest list for multiple platform.
-	manifest := ocispec.Manifest{}
-	if err = json.Unmarshal(bytes, &manifest); err != nil {
-		return err
+	for _, manifest := range manifestIndex.Manifests {
+		mDesc := ocispec.Descriptor{
+			MediaType: manifest.MediaType,
+			Digest:    manifest.Digest,
+			Size:      manifest.Size,
+		}
+		mir, err := fetcher.Fetch(ctx, mDesc)
+		if err != nil {
+			return nil, errors.Wrap(err, "fetch remote cache")
+		}
+		manifestBytes, err := io.ReadAll(mir)
+		if err != nil {
+			return nil, errors.Wrap(err, "read remote cache bytes to manifest")
+		}
+		if err = content.WriteBlob(ctx, pvd.ContentStore(), ref, bytes.NewReader(manifestBytes), mDesc); err != nil {
+			return nil, errors.Wrap(err, "remote cache write blob failed")
+		}
+
 	}
 
+	matchDescs, err := utils.GetManifests(ctx, pvd.ContentStore(), *manifestIndexDesc, d.platformMC)
+	if err != nil {
+		return nil, errors.Wrap(err, "get remote cache manifest list")
+	}
+	if sourceDesc.Platform == nil {
+		sourceDesc.Platform = &ocispec.Platform{
+			Architecture: runtime.GOARCH,
+			OS:           runtime.GOOS,
+		}
+	}
+	targetManifest := ocispec.Manifest{}
 	cs := pvd.ContentStore()
-	for _, layer := range manifest.Layers {
-		if _, err := cs.Update(ctx, content.Info{
-			Digest: layer.Digest,
-			Size:   layer.Size,
-			Labels: layer.Annotations,
-		}); err != nil {
-			if errors.Is(err, containerdErrDefs.ErrNotFound) {
-				return errdefs.ErrNotSupport
+	for _, desc := range matchDescs {
+		if reflect.DeepEqual(desc.Platform, sourceDesc.Platform) {
+			targetManifestBytes, err := content.ReadBlob(ctx, pvd.ContentStore(), desc)
+			if err != nil {
+				return nil, errors.Wrap(err, "read remote cache manifest")
 			}
-			return errors.Wrap(err, "update cache layer")
+			if err = json.Unmarshal(targetManifestBytes, &targetManifest); err != nil {
+				return nil, err
+			}
+			for _, layer := range targetManifest.Layers {
+				if _, err := cs.Update(ctx, content.Info{
+					Digest: layer.Digest,
+					Size:   layer.Size,
+					Labels: layer.Annotations,
+				}); err != nil {
+					if errors.Is(err, containerdErrDefs.ErrNotFound) {
+						return nil, errdefs.ErrNotSupport
+					}
+					return nil, errors.Wrap(err, "update cache layer")
+				}
+			}
+			return manifestIndexDesc, nil
 		}
 	}
-	return nil
+	return nil, nil
 }
 
 // PushRemoteCache update cache manifest and push to remote
-func (d *Driver) PushRemoteCache(ctx context.Context, pvd accelcontent.Provider, ref string) error {
+func (d *Driver) PushRemoteCache(ctx context.Context, pvd accelcontent.Provider, ref string, sourceDesc, cacheIndexDesc *ocispec.Descriptor) error {
```

Comment:
We always treat the cache image as a manifest index? Can we also adapt the manifest?

Reply:
ok